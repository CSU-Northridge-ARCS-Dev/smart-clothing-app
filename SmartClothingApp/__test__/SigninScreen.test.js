//jest.useFakeTimers()

import React from 'react';
import { Alert } from 'react-native';
import { act } from 'react-test-renderer';
import { Text, TextInput, ScrollView } from 'react-native';
import { HelperText } from 'react-native-paper';
import renderer from 'react-test-renderer';
import { Button } from 'react-native-paper';
import { waitFor } from '@testing-library/react-native';
import SigninScreen from '../src/screens/SigninScreen/index.jsx';
import { useDispatch } from 'react-redux';
import { startLoginWithEmail } from '../src/actions/userActions.js';



// Mock dependencies or Redux actions as needed
jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));
// mocks userActions.js module 
//    functions: setLoginWithEmail and SetAuthError 
jest.mock('../src/actions/userActions.js', () => ({
  startLoginWithEmail: jest.fn(),
  setAuthError: jest.fn(),
}));

// Mock userActions.js module functions: startLoginWithEmail and setAuthError
jest.mock('../src/actions/userActions.js', () => ({
  startLoginWithEmail: jest.fn(),
  setAuthError: jest.fn(),
}));

// Mock userActions.js module functions: startLoginWithEmail and setAuthError
jest.mock('../src/actions/userActions.js', () => ({
  startLoginWithEmail: jest.fn().mockResolvedValueOnce(), // Mock to resolve (successful sign-in)
  setAuthError: jest.fn(),
}));



// Test Suite
describe('SigninScreen Error Alerts', () => {
  // renders SigninScreen with mock navigation prop and get instance of component's rendered tree
  let component;
  let instance;

  // Render SigninScreen with mock navigation prop and get the instance of the component's rendered tree
  beforeEach(() => {
    component = renderer.create(<SigninScreen navigation={{ navigate: jest.fn() }} />);
    instance = component.root;
  });

  // Clean up after each test by resetting the alert spy
  afterEach(() => {
    jest.clearAllMocks();
  });



  // Test Case 1: if component renders correctly
  it('renders correctly', () => {
    // renders SigninScreen with mocked navigation prop and converts it to JSON to save Snapshot
    const tree = renderer.create(<SigninScreen navigation={{ navigate: jest.fn() }} />).toJSON();
    expect(tree).toMatchSnapshot(); // compares last Snapshot to detect changes in component code
  });



  // Test Case 2: 
  it('displays HelperText and Alert error for an invalid email input', async () => {
    // Find the TextInput components based on their labels
    let emailInput = instance.findByProps({ label: 'Email' });
    let passwordInput = instance.findByProps({ label: 'Password' });
    
    // Simulate user input for email and password
    await waitFor(() => {
      emailInput.props.onChangeText('invalid-email'); 
      passwordInput.props.onChangeText('password123');
    });

    // Track for any Alerts
    jest.spyOn(Alert, 'alert');
    // Simulate pressing sign in button
    act(() => {
      let button = instance.findByProps({ ID: 'signInButton' });
      button.props.onPress();
    });

    // Ensure expected alert error message is displayed
    expect(Alert.alert).toHaveBeenCalledWith("Authentication Error", 
    "Please correct the following errors:\n\n" + 
    "Enter valid email.\n"+""
      ); 


    // Ensure appropriate error message is displayed in HelperText
    const helperTextElement = instance.findByType(HelperText);
    const errorElements = instance.findAllByProps({ type: 'error' });
    const firstErrorElement = errorElements[0];
    const firstEmailErrorText = firstErrorElement.props.children;

    // Password Text Field doesn't have HelperText anymore 
    // const secondErrorElement = errorElements[1];
    // const secondEmailErrorText = secondErrorElement.props.children;
    

    expect(firstEmailErrorText).toEqual('Email is invalid!');
    expect(helperTextElement.props.visible).toBe(true);
    //expect(secondEmailErrorText).toEqual('');
  });

   

  // Test Case 3:
  it('displays HelperText and Alert error for an invalid empty password input', async() => {
    let emailInput = instance.findByProps({ label: 'Email' });
    let passwordInput = instance.findByProps({ label: 'Password' });
    
    await waitFor(() => {
      emailInput.props.onChangeText("Example@gmail.com");
      passwordInput.props.onChangeText('');
    });

    jest.spyOn(Alert, 'alert');
    act(() => {
      let button = instance.findByProps({ ID: 'signInButton' });
      button.props.onPress();
    });

    expect(Alert.alert).toHaveBeenCalledWith("Authentication Error", 
    "Please correct the following errors:\n\n" + ""+
    "Password cannot be empty"
    ); 
    
    // Ensure appropriate error message is displayed in HelperText
    const helperTextElement = instance.findByType(HelperText);
    const errorElements = instance.findAllByProps({ type: 'error' });
    const firstErrorElement = errorElements[0];
    const firstEmailErrorText = firstErrorElement.props.children;

    // Password Text Field doesn't have HelperText anymore 
    // const secondErrorElement = errorElements[1];
    // const secondEmailErrorText = secondErrorElement.props.children;
    
    expect(firstEmailErrorText).toEqual('Email is invalid!');
    expect(helperTextElement.props.visible).toBe(false);
    //expect(secondEmailErrorText).toEqual('Password cannot be empty');
  });



  // Test Case 4:
  it('displays HelperText and Alert error for an invalid passowrd below 6 characters', async () => {
    let emailInput = instance.findByProps({ label: 'Email' });
    let passwordInput = instance.findByProps({ label: 'Password' });

    await waitFor(() => {
      emailInput.props.onChangeText("Example@gmail.com");
      passwordInput.props.onChangeText('short');
    });

    jest.spyOn(Alert, 'alert');
    act(() => {
       let button = instance.findByProps({ ID: 'signInButton' });
       button.props.onPress();
    });

    expect(Alert.alert).toHaveBeenCalledWith("Authentication Error", 
    "Please correct the following errors:\n\n" + ""+
    "Password length cannot be less than 6."
    ); 

    // Ensure appropriate error message is displayed in HelperText
    const helperTextElement = instance.findByType(HelperText);
    const errorElements = instance.findAllByProps({ type: 'error' });
    const firstErrorElement = errorElements[0];
    const firstEmailErrorText = firstErrorElement.props.children;

    // Password Text Field doesn't have HelperText anymore 
    // const secondErrorElement = errorElements[1];
    // const secondEmailErrorText = secondErrorElement.props.children;

    expect(firstEmailErrorText).toEqual('Email is invalid!');
    expect(helperTextElement.props.visible).toBe(false);
    //expect(secondEmailErrorText).toEqual('Password cannot be empty');
  });



  // Test Case 5:
  it('displays HelperText and Alert error for both invalid passowrd and invalid email', async () => {
    let emailInput = instance.findByProps({ label: 'Email' });
    let passwordInput = instance.findByProps({ label: 'Password' });
    
    await waitFor(() => {
      emailInput.props.onChangeText("Invalid-Email");
      passwordInput.props.onChangeText('short');
    });

    jest.spyOn(Alert, 'alert');
    act(() => {
       let button = instance.findByProps({ ID: 'signInButton' });
       button.props.onPress();
    });

    expect(Alert.alert).toHaveBeenCalledWith("Authentication Error", 
    "Please correct the following errors:\n\n" + 
    "Enter valid email.\n" +
    "Password length cannot be less than 6."
    ); 

    // Ensure appropriate error message is displayed in HelperText
    const helperTextElement = instance.findByType(HelperText);
    const errorElements = instance.findAllByProps({ type: 'error' });
    const firstErrorElement = errorElements[0];
    const firstEmailErrorText = firstErrorElement.props.children;

    // Password Text Field doesn't have HelperText anymore 
    // const secondErrorElement = errorElements[1];
    // const secondEmailErrorText = secondErrorElement.props.children;

    expect(firstEmailErrorText).toEqual('Email is invalid!');
    expect(helperTextElement.props.visible).toBe(true);
    //expect(secondEmailErrorText).toEqual('Password cannot be empty');
  });

// Test Case 6: Successful sign-in without errors
it('successfully signs in without errors', async () => {
  // Find the TextInput components based on their labels
  let emailInput = instance.findByProps({ label: 'Email' });
  let passwordInput = instance.findByProps({ label: 'Password' });
  
  // Simulate user input for email and password
  await waitFor(() => {
    emailInput.props.onChangeText('valid@gmail.com'); 
    passwordInput.props.onChangeText('password123');
  });

  // Ensure that Alert.alert is not called (no errors expected)
  expect(Alert.alert).not.toHaveBeenCalled();  
});  

//Test Case 7:
it('navigates to Forgot Password screen when "Forgot" button is pressed', async () => {
  // Mock the navigation object
  const mockNavigate = jest.fn();
  const navigation = { navigate: mockNavigate };

  // Render the component with the mocked navigation prop
  component = renderer.create(<SigninScreen navigation={navigation} />);
  instance = component.root;

  // Find the "Forgot" button by its label
  const forgotButton = instance.findByProps({ children: "Forgot your Username/Password ?" });

  // Simulate pressing the "Forgot" button
  act(() => {
    forgotButton.props.onPress();
  });

  // Ensure that navigation.navigate was called with the correct screen name
  expect(mockNavigate).toHaveBeenCalledWith('Forgot');
});


//Testcase 8
it('successfully signs in without errors', async () => {
  // Mock the navigation object
  const mockNavigate = jest.fn();
  const navigation = { navigate: mockNavigate };

  // Mock the useDispatch function
  const mockDispatch = jest.fn();
  const mockStartLoginWithEmail = jest.fn(); // Mock the startLoginWithEmail function

  // Set up useDispatch to return the mockDispatch function
  useDispatch.mockReturnValueOnce(mockDispatch);

  // Render the component with the mocked navigation prop
  component = renderer.create(<SigninScreen navigation={navigation} />);
  instance = component.root;

  // Find the TextInput components based on their labels
  const emailInput = instance.findByProps({ label: 'Email' });
  const passwordInput = instance.findByProps({ label: 'Password' });

  // Simulate user input for email and password
  await act(async () => {
    emailInput.props.onChangeText('valid@gmail.com');
    passwordInput.props.onChangeText('password123');
  });

  // Find the "Sign In" button
  const signInButton = instance.findByProps({ ID: 'signInButton' });

  // Mock the startLoginWithEmail function to resolve (successful sign-in)
  startLoginWithEmail.mockResolvedValueOnce();

  // Simulate pressing the "Sign In" button
  await act(async () => {
    signInButton.props.onPress();
  });

  // Ensure that navigation.navigate was not called (no navigation after successful sign-in)
  expect(mockNavigate).not.toHaveBeenCalled();

});




  // Test Case 6: Successful sign-in without errors
  it('successfully signs in without errors', async () => {
    // Find the TextInput components based on their labels
    let emailInput = instance.findByProps({ label: 'Email' });
    let passwordInput = instance.findByProps({ label: 'Password' });
    // Simulate user input for email and password
    await waitFor(() => {
      emailInput.props.onChangeText('valid@gmail.com');
      passwordInput.props.onChangeText('password123');
    });
    // Ensure that Alert.alert is not called (no errors expected)
    expect(Alert.alert).not.toHaveBeenCalled();
  });
});
