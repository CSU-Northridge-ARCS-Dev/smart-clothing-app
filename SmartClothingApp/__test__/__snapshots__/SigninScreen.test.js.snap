// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SigninScreen Error Alerts renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgb(17, 96, 164)",
          "height": 521.09375,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": 521.09375,
              "width": undefined,
            },
            {
              "opacity": 0.2,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          {
            "height": 125.0625,
            "resizeMode": "contain",
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "rgb(253, 252, 255)",
          "borderTopLeftRadius": 25,
          "borderTopRightRadius": 25,
          "paddingHorizontal": 20,
          "paddingTop": 25,
          "transform": [
            {
              "translateY": -25,
            },
          ],
        }
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "ChakraPetch_BoldItalic",
              "fontSize": 28,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 36,
            },
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "writingDirection": "ltr",
            },
            [
              {
                "borderLeftColor": "rgb(17, 96, 164)",
                "borderLeftWidth": 10.416666666666668,
                "color": "rgb(17, 96, 164)",
                "fontFamily": "ChakraPetch_BoldItalic",
                "fontSize": 28,
                "paddingLeft": 10,
              },
              {
                "marginBottom": 20.84375,
              },
            ],
          ]
        }
      >
        Sign In
      </Text>
      <Text
        style={
          [
            {
              "fontFamily": "Poppins_Bold",
              "fontSize": 16,
              "fontWeight": "500",
              "letterSpacing": 0.15,
              "lineHeight": 24,
            },
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "writingDirection": "ltr",
            },
            [
              {
                "color": "rgb(17, 96, 164)",
                "fontFamily": "Poppins_Bold",
                "fontSize": 16,
              },
              {
                "marginBottom": 20.84375,
              },
            ],
          ]
        }
      >
        Welcome back. Login to continue
      </Text>
      <View>
        <View>
          <View
            style={{}}
          >
            <View
              pointerEvents="none"
              style={
                [
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  {
                    "backgroundColor": "rgba(255, 251, 254, 1)",
                    "borderColor": "rgba(121, 116, 126, 1)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                  undefined,
                ]
              }
              testID="text-input-outline"
            />
            <View>
              <View
                style={
                  [
                    {
                      "paddingBottom": 0,
                    },
                    {
                      "minHeight": 56,
                      "paddingTop": 8,
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  pointerEvents="none"
                  style={
                    {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                      ],
                      "zIndex": 3,
                    }
                  }
                >
                  <Text
                    collapsable={false}
                    maxFontSizeMultiplier={1.5}
                    numberOfLines={1}
                    style={
                      {
                        "backgroundColor": "rgba(255, 251, 254, 1)",
                        "color": "transparent",
                        "fontFamily": "Poppins_Regular",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 8,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "opacity": -0.6666666666666667,
                        "paddingHorizontal": 0,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 33,
                        "transform": [
                          {
                            "translateX": 0,
                          },
                          {
                            "translateY": 0,
                          },
                          {
                            "scale": 1,
                          },
                          {
                            "scaleY": 1,
                          },
                        ],
                        "width": -16,
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    Email
                  </Text>
                  <Text
                    collapsable={false}
                    maxFontSizeMultiplier={1.5}
                    numberOfLines={1}
                    onLayout={[Function]}
                    style={
                      {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontFamily": "Poppins_Regular",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "opacity": 0,
                        "paddingHorizontal": 16,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 32,
                        "transform": [
                          {
                            "translateX": 0,
                          },
                          {
                            "translateY": 0,
                          },
                          {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                    testID="text-input-outlined-label-active"
                  >
                    Email
                  </Text>
                  <Text
                    collapsable={false}
                    maxFontSizeMultiplier={1.5}
                    numberOfLines={1}
                    style={
                      {
                        "color": "rgba(73, 69, 79, 1)",
                        "fontFamily": "Poppins_Regular",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "opacity": 0,
                        "paddingHorizontal": 16,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 32,
                        "transform": [
                          {
                            "translateX": 0,
                          },
                          {
                            "translateY": 0,
                          },
                          {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                    testID="text-input-outlined-label-inactive"
                  >
                    Email
                  </Text>
                </View>
                <TextInput
                  editable={true}
                  maxFontSizeMultiplier={1.5}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder=" "
                  placeholderTextColor="rgba(73, 69, 79, 1)"
                  selectionColor="rgba(103, 80, 164, 1)"
                  style={
                    [
                      {
                        "margin": 0,
                        "zIndex": 1,
                      },
                      {
                        "height": 48,
                      },
                      {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "Poppins_Regular",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      [
                        {},
                      ],
                      undefined,
                    ]
                  }
                  testID="text-input-outlined"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
        </View>
        <Text
          collapsable={false}
          maxFontSizeMultiplier={1.5}
          onLayout={[Function]}
          style={
            {
              "color": "rgba(179, 38, 30, 1)",
              "fontFamily": "Poppins_Regular",
              "fontSize": 12,
              "fontWeight": "400",
              "letterSpacing": 0.25,
              "lineHeight": 20,
              "opacity": 0,
              "paddingHorizontal": 12,
              "paddingVertical": 4,
              "textAlign": "left",
              "transform": [],
              "writingDirection": "ltr",
            }
          }
        >
          Email is invalid!
        </Text>
      </View>
      <View>
        <View
          style={{}}
        >
          <View
            pointerEvents="none"
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                {
                  "backgroundColor": "rgba(255, 251, 254, 1)",
                  "borderColor": "rgba(121, 116, 126, 1)",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
                undefined,
              ]
            }
            testID="text-input-outline"
          />
          <View>
            <View
              style={
                [
                  {
                    "paddingBottom": 0,
                  },
                  {
                    "minHeight": 56,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                    ],
                    "zIndex": 3,
                  }
                }
              >
                <Text
                  collapsable={false}
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  style={
                    {
                      "backgroundColor": "rgba(255, 251, 254, 1)",
                      "color": "transparent",
                      "fontFamily": "Poppins_Regular",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 8,
                      "letterSpacing": 0.15,
                      "lineHeight": undefined,
                      "opacity": -0.6666666666666667,
                      "paddingHorizontal": 0,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 33,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateY": 0,
                        },
                        {
                          "scale": 1,
                        },
                        {
                          "scaleY": 1,
                        },
                      ],
                      "width": -16,
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Password
                </Text>
                <Text
                  collapsable={false}
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    {
                      "color": "rgba(103, 80, 164, 1)",
                      "fontFamily": "Poppins_Regular",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "letterSpacing": 0.15,
                      "lineHeight": undefined,
                      "opacity": 0,
                      "paddingHorizontal": 16,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 32,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateY": 0,
                        },
                        {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                  testID="text-input-outlined-label-active"
                >
                  Password
                </Text>
                <Text
                  collapsable={false}
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  style={
                    {
                      "color": "rgba(73, 69, 79, 1)",
                      "fontFamily": "Poppins_Regular",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "letterSpacing": 0.15,
                      "lineHeight": undefined,
                      "opacity": 0,
                      "paddingHorizontal": 16,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 32,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateY": 0,
                        },
                        {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                  testID="text-input-outlined-label-inactive"
                >
                  Password
                </Text>
              </View>
              <TextInput
                editable={true}
                maxFontSizeMultiplier={1.5}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder=" "
                placeholderTextColor="rgba(73, 69, 79, 1)"
                secureTextEntry={true}
                selectionColor="rgba(103, 80, 164, 1)"
                style={
                  [
                    {
                      "margin": 0,
                      "zIndex": 1,
                    },
                    {
                      "height": 48,
                    },
                    {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "Poppins_Regular",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "letterSpacing": 0.15,
                      "lineHeight": undefined,
                      "paddingHorizontal": 16,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    [
                      {},
                    ],
                    undefined,
                  ]
                }
                testID="text-input-outlined"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
        <Text
          allowFontScaling={false}
          onPress={[Function]}
          selectable={false}
          style={
            [
              {
                "color": "black",
                "fontSize": 25,
              },
              {
                "position": "absolute",
                "right": 10,
                "top": 16,
                "transform": [
                  {
                    "translateY": 0,
                  },
                ],
              },
              {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {
                "fontWeight": "700",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View>
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 20,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="button-container-outer-layer"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderRadius": 20,
                "borderStyle": "solid",
                "borderWidth": 0,
                "flex": undefined,
                "minWidth": 64,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="button-container"
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "borderRadius": 20,
                  },
                ]
              }
              testID="button"
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "fontFamily": "Poppins_Regular",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        false,
                        {
                          "marginHorizontal": 12,
                        },
                        undefined,
                        false,
                        {
                          "color": "rgba(103, 80, 164, 1)",
                          "fontFamily": "Poppins_Regular",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        undefined,
                      ],
                    ]
                  }
                  testID="button-text"
                >
                  Forgot your Username/Password ?
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginVertical": 10,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(28, 27, 31, 0.12)",
            "borderRadius": 20,
            "flex": 2,
            "marginHorizontal": 20.833333333333336,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container-outer-layer"
      >
        <View
          ID="signInButton"
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(28, 27, 31, 0.12)",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flex": undefined,
              "minWidth": 64,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="button-container"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 20,
                },
              ]
            }
            testID="button"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "fontFamily": "Poppins_Regular",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "rgba(28, 27, 31, 0.38)",
                        "fontFamily": "Poppins_Regular",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                  ]
                }
                testID="button-text"
              >
                Sign In
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderRadius": 20,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 20,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "fontFamily": "Poppins_Regular",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    false,
                    {
                      "marginHorizontal": 12,
                    },
                    undefined,
                    false,
                    {
                      "color": "rgba(103, 80, 164, 1)",
                      "fontFamily": "Poppins_Regular",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
              testID="button-text"
            >
              Don't have an account? Sign Up
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
